bin/update-ips.bash

```diff
@@ -161,8 +161,8 @@ get_dns_ips() {
   local -a ips
   mapfile -t ips < <(dig +short "${domain}" | grep '^[.0-9]*$')
   if [ ${#ips[@]} -eq 0 ]; then
-    log_error "No IPs for ${domain} was found"
-    exit 1
+    log_error "No IPs for ${domain} was found. Will block all IPs"
+    echo "0.0.0.0"
```

Comment:
Will this not the opposite, i.e. allow all IPs? This function is used to resolve a domain name to an IP so that it can be added to the allow list, no?

Reply:
This eventually sets the CIDR in the NP to 0.0.0.0/32, i.e. just the 0.0.0.0 ip which would never be a real IP meaning it blocks everything.

Reply:
The `process_ips_to_cidrs` seems to be the one adding the `/32`

Reply:
Ah of course :facepalm: 

I'm still a bit worried that we might at some point rewrite this and not all IPs go through that same IP to CIDR transformation function.

Also, why would you want it not to fail if the domain can't be resolved? Is it not better to become aware of it?

Reply:
So, when you use a LoadBalancer Service for ingress-nginx you can't create the domains before installing ingress-nginx. And before you could not run update-ips since it would fail and the schema validation would complain if you tried to apply anything. This seemed like a better approach then having to manually set all NPs to something first. Not perfect but I'm not sure if there is a better solution.

Although I just realized I forgot to mention you would have to re-run update-ips again after setting up the DNS.

Reply:
Could it not be left emptied instead to disallow everything that way? :thinking:
(IIRC the associated rules _shouldn't_ be rendered in that case, because empty netpols allow everything, but we have a split between the old and new netpol still.)

Reply:
Maybe split the script change and the documentation changes into separate PRs, both good for changelog purposes and review purposes

Reply:
> Could it not be left emptied instead to disallow everything that way? 🤔

If I leave it empty instead the CIDR ends up as just `/32` which fails in the apply